src/inversion_scripts/jacobian.py

```diff
@@ -449,173 +689,100 @@ def apply_tropomi_operator(
-            polygon_geoschem = Polygon(
-                np.column_stack((geoschem_corners_lon, geoschem_corners_lat))
-            )
-            # Calculate overlapping area as the intersection of the two polygons
-            if polygon_geoschem.intersects(polygon_tropomi):
-                overlap_area[gridcellIndex] = polygon_tropomi.intersection(
-                    polygon_geoschem
-                ).area
-
-        # If there is no overlap between GEOS-Chem and TROPOMI, skip to next observation:
-        if sum(overlap_area) == 0:
-            continue
-
-        # =======================================================
-        #       Map GEOS-Chem to TROPOMI observation space
-        # =======================================================
-
         # Otherwise, initialize tropomi virtual xch4 and virtual sensitivity as zero
         area_weighted_virtual_tropomi = 0  # virtual tropomi xch4
         area_weighted_virtual_tropomi_sensitivity = 0  # virtual tropomi sensitivity
 
         # For each GEOS-Chem grid cell that touches the TROPOMI pixel:
-        for gridcellIndex in range(len(gc_coords)):
-
-            # Get GEOS-Chem lat/lon indices for the cell
-            iGC, jGC = ij_GC[gridcellIndex]
-
-            # Get GEOS-Chem pressure edges for the cell
-            p_gc = GEOSCHEM["PEDGE"][iGC, jGC, :]
-
-            # Get GEOS-Chem methane for the cell
-            gc_CH4 = GEOSCHEM["CH4"][iGC, jGC, :]
-
-            # Get merged GEOS-Chem/TROPOMI pressure grid for the cell
-            merged = merge_pressure_grids(p_sat, p_gc)
-
-            # Remap GEOS-Chem methane to TROPOMI pressure levels
-            sat_CH4 = remap(
-                gc_CH4,
+        # Get GEOS-Chem pressure edges for the cell
+        p_gc = GEOSCHEM["PEDGE"][gridcell_dict["iGC"], gridcell_dict["jGC"], :]
+        # Get GEOS-Chem methane for the cell
+        gc_CH4 = GEOSCHEM["CH4"][gridcell_dict["iGC"], gridcell_dict["jGC"], :]
+        # Get merged GEOS-Chem/TROPOMI pressure grid for the cell
+        merged = merge_pressure_grids(p_sat, p_gc)
+        # Remap GEOS-Chem methane to TROPOMI pressure levels
+        sat_CH4 = remap(
+            gc_CH4,
+            merged["data_type"],
+            merged["p_merge"],
+            merged["edge_index"],
+            merged["first_gc_edge"],
+        )  # ppb
+        # Convert ppb to mol m-2
+        sat_CH4_molm2 = sat_CH4 * 1e-9 * dry_air_subcolumns  # mol m-2
+        # Derive the column-averaged XCH4 that TROPOMI would see over this ground cell
+        # using eq. 46 from TROPOMI Methane ATBD, Hasekamp et al. 2019
+        virtual_tropomi_gridcellIndex = (
+            sum(apriori + avkern * (sat_CH4_molm2 - apriori))
+            / sum(dry_air_subcolumns)
+            * 1e9
+        )  # ppb
+        # Weight by overlapping area (to be divided out later) and add to sum
+        area_weighted_virtual_tropomi += (
+            gridcell_dict["overlap_area"] * virtual_tropomi_gridcellIndex
```

Comment:
Re: my comment below, we don't need this line, for example, since we already did the area weighting in constructing `gridcell_dicts` and further down end up dividing out `gridcell_dict["overlap_area"]`